pkg/providers/cloudstack/cloudstack.go

```diff
@@ -63,7 +63,7 @@ type cloudstackProvider struct {
 	selfSigned            bool
 	templateBuilder       *CloudStackTemplateBuilder
 	skipIpCheck           bool
-	validator             *Validator
+	validators            map[string]*Validator
```

Comment:
here too - why not make it a list? It seems a bit simpler that way in my mind. Is there any specific usecase you have in mind for making it a map? It just feels like more work for the user to maintain this mapping in my mind but open to other opinions

Reply:
Never mind, I think a map is the right move so that we can match these Profiles to an Availability Zone for preflight checking